docs/architecture/adr-002-event-manager-contract.md

```diff
@@ -0,0 +1,95 @@
+# ADR 002: POAP Manager Contract
+
+## Changelog
+
+- July 11, 2022: Initial draft started;
+- July 12, 2022: Initial draft finished.
+
+## Status
+DRAFTED
+
+## Abstract
+This ADR defines the architecture of the POAP Manager contract. This contract give a user the possibility to manage
+the minting process of a POAP by setting a range of conditions that users needs to fulfill in order to be able to get
+the POAP.
+
+## Context
+Inside Desmos based social networks, users have a profile with which they interact with the different features that the
+applications provides. From the `v4.1.0` release, users can also create contents on-chain through the post module. Looking
+forward to the next Cosmoverse 2022, we would like to be able to check the eligibility to mint a POAP. Attendees that wants
+to receive the badge, will be required to fulfill three simple tasks:
+1. Create a post
+2. Comment a post
+3. React to a post
+
+Once done, the contract will trigger the mint of the POAP automatically. The mint will take place separately inside the
+`POAP-Contract` and no one else except the `POAP-Manager` will be able to perform the operation.
+
+## Decision
+The idea is to write a simple contract that make usage of the recently tagged `v1.0.0` of the Desmos-Bindings package
+to check on the 3 mint conditions and then call the `POAP-contract` to mint the POAP.
+
+## Specifications
+Here below the specifications for the contract's messages:
+
+### Messages
+
+#### Instantiate
+```rust
+pub struct InstantiateMsg {
+  pub admin: Addr,
+  pub poap_contract: Addr,
+  pub subspace_id: u64,
+}
+```
+
+* The `admin` identifies the user that controls the contract;
+* The `poap_contract` that the event manager control;
+* The `subspace_id` identifies the dApp where the contract lives;
+
+#### Execute
+```rust
+pub enum ExecuteMsg{
+  TryMint{},
```

Comment:
I did not add the `subspace_id` field to the message because it's part of the contract state and initialised with the contract `InstatiateMsg`. This to force the checks on a particular subspace to avoid users looking for created posts and comments and likes in other subspaces. Why this? Because my original idea was to check if a user has made any posts, reactions, comment and let them mint the poap after that checks.
Sadly, Desmos still not support the query of posts by users, so that would be an additional work for the contract and I think it will increase the costs of computation quite a lot. Looking at your suggestion I thought that, to simplify this, we can make them comment and react only 1 specific post (so we can pass the id to chain). The problem will still subsist if we require them also to create a post but, since a comment is a post, I think we can rely only on those two checks.